src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -839,7 +839,7 @@ public static int compare(final String str1, final String str2) {
      * @since 3.5
      */
     public static int compare(final String str1, final String str2, final boolean nullIsLess) {
-        if (str1 == str2) { // NOSONARLINT this intentionally uses == to allow for both null
+        if (Objects.equals(str1, str2)) { // str1 and str2 as equal if both are null or identical
```

Comment:
-1: You're circumventing code someone toke care to optimize and comment: "this intentionally uses == to allow for both null"!